### ISSUE TITLE
Fatal Error: Upgrading phpseclib/phpseclib (3.0.21 => 3.0.22) 

### ISSUE THREAD

**Issue by parallels999** [phpseclib/phpseclib#1941]
Does anyone else have this problem?
I'm trying to update, and it seems like it's a breaking change
```
Fatal error: 
  Uncaught UnexpectedValueException: 
    JIT on Windows is not currently supported in \vendor\phpseclib\phpseclib\phpseclib\bootstrap.php:28
    Stack trace:
      #0 \vendor\composer\autoload_real.php(41): require()
      #1 \vendor\composer\autoload_real.php(45): {closure}('decc78cc4436b12...', 'C:\\laravel\\ssw\\...') 
      #2 \vendor\autoload.php(25): ComposerAutoloaderInite3b2492fbaae1a014a5832dc98cf3829::getLoader()
```
On **3.0.21** works, How should I proceed?

**Comment by parallels999** [comment]
On `php.ini` I add
```
opcache.jit=off
```
It is working, but I don't know if it's the right answer

**Comment by terrafrost** [comment]
It may technically be a BC breaking change but the problem is that, on Windows, it turns out that [JIT _is_ broken](https://github.com/php/php-src/issues/11917). Sure, I've only ever seen issues when the PHP64 engine is used but, the problem is, if there are issues when that engine is being used then I can't be sure that other problems aren't lurking beneath the surface just waiting for someone to find and I don't want people expecting me to fix issues when the issues have nothing to do with phpseclib anyway. I mean, I suppose I could write a new engine just for JIT enabled Windows PHP installs but without really having a good handle on what is causing the problem, anyway, (well, aside from the fact that I know the problem is caused by PHP lol), I can't guarantee that problems could still exist in whatever rewrite I do.

In spite of that, if you want to use JIT on Windows, you can do `define('PHPSECLIB_ALLOW_JIT', true);` at the top of your file.

**Comment by parallels999** [comment]
Thanks for responding, `define('PHPSECLIB_ALLOW_JIT', true);` seems like a good option, but this is just a test environment, I can disable it, it doesn't matter

**Comment by rossaddison** [comment]
Here is my stack trace: 

````
C:\wamp64\www\yii3-i-4>php ./vendor/bin/psalm
Uncaught UnexpectedValueException: JIT on Windows is not currently supported in C:\wamp64\www\yii3-i-4\vendor\phpseclib\phpseclib\phpseclib\bootstrap.php:28
Stack trace:
#0 C:\wamp64\www\yii3-i-4\vendor\composer\autoload_real.php(41): require()
#1 C:\wamp64\www\yii3-i-4\vendor\composer\autoload_real.php(45): ComposerAutoloaderInitbde380415d49cb0ecf0e4578ac588a67::{closure}('decc78cc4436b12...', 'C:\\wamp64\\www\\y...')
#2 C:\wamp64\www\yii3-i-4\vendor\autoload.php(25): ComposerAutoloaderInitbde380415d49cb0ecf0e4578ac588a67::getLoader()
#3 C:\wamp64\www\yii3-i-4\vendor\vimeo\psalm\src\Psalm\Internal\CliUtils.php(132): require_once('C:\\wamp64\\www\\y...')
#4 C:\wamp64\www\yii3-i-4\vendor\vimeo\psalm\src\Psalm\Internal\Cli\Psalm.php(233): Psalm\Internal\CliUtils::requireAutoloaders('C:\\wamp64\\www\\y...', false, 'vendor')
#5 C:\wamp64\www\yii3-i-4\vendor\vimeo\psalm\src\Psalm\Internal\IncludeCollector.php(35): Psalm\Internal\Cli\Psalm::Psalm\Internal\Cli\{closure}()
#6 C:\wamp64\www\yii3-i-4\vendor\vimeo\psalm\src\Psalm\Internal\Cli\Psalm.php(229): Psalm\Internal\IncludeCollector->runAndCollect(Object(Closure))
#7 C:\wamp64\www\yii3-i-4\vendor\vimeo\psalm\psalm(9): Psalm\Internal\Cli\Psalm::run(Array)
#8 C:\wamp64\www\yii3-i-4\vendor\bin\psalm(120): include('C:\\wamp64\\www\\y...')
#9 {main}
(Psalm (unknown version) crashed due to an uncaught Throwable)
````

**Comment by rossaddison** [comment]
Psalm 5.15.0@5c774aca4746caf3d239d9c8cadb9f882ca29352

**Comment by rossaddison** [comment]
![phpseclib](https://github.com/phpseclib/phpseclib/assets/8538339/d08153b6-ec3c-4ded-be6d-33d0205a738a)

This is my current wampserver setup.

**Comment by rossaddison** [comment]
I am currently reading the bootstrap.php file and get the following:

````
// see https://github.com/php/php-src/issues/11917
if (strtoupper(substr(PHP_OS, 0, 3)) === 'WIN' && function_exists('opcache_get_status') && !defined('PHPSECLIB_ALLOW_JIT')) {
    $status = opcache_get_status();
    if ($status && isset($status['jit']) && $status['jit']['enabled'] && $status['jit']['on']) {
        throw new UnexpectedValueException(
            'JIT on Windows is not currently supported'
        );
    }
}
````

**Comment by terrafrost** [comment]
Like the error message says, JIT on Windows is not supported. This is because JIT on Windows is buggy:

https://github.com/php/php-src/issues/11917

Consequently, running JIT on Windows is rather like letting making it so that `$a == $b` didn't _actually_ return whether or not `$a == $b` but instead just returned `true` or `false` at _random_. I mean, maybe it'll return the correct answer or maybe it won't. But I wouldn't make use of anything so unpredictable and unreliable and, per the above ticket, I view JIT on Windows as being that unreliable.

If, none-the-less, you want to run JIT on Windows, you can do `define('PHPSECLIB_ALLOW_JIT', true);`, however, if you do that, do not expect any support from me if you run into further issues later on down the line.

**Comment by rossaddison** [comment]
Thank you for your quick response. Apologies for not getting back sooner. I did a recent update with composer and encountered this problem. So I am hoping it will sort itself out. I have not intentionally endeavoured to run JIT so it is a bit odd that this has crept in. 

I ran the 

````
use Yiisoft\VarDumper\VarDumper;
echo VarDumper::dump(opcache_get_status()['jit']);
````

at the bottom of the resources\views\layout\invoice.php file ...
````
[
    'enabled' => false
    'on' => false
    'kind' => 0
    'opt_level' => 0
    'opt_flags' => 0
    'buffer_size' => 0
    'buffer_free' => 0
]
````
which as also indicated by the screenshot is not being run under php 8.2.10.

Fortunately I make regular backups so hopefully this will iron itself out.

Thanks for your response.
